src/backend/storeManagers/nile/library.ts

```diff
@@ -0,0 +1,483 @@
+      )
+    }
+  }
+}
+
+/**
+ * Fetches and parses the game's `fuel.json` file
+ */
+export function fetchFuelJSON(
+  appName: string,
+  installedPath?: string
+): FuelSchema | null {
+  const game = getGameInfo(appName)
+  const basePath = installedPath ?? game?.install.install_path
+  if (!basePath) {
+    logError(['Could not find install path for', appName], LogPrefix.Nile)
+    return null
+  }
+
+  const fuelJSONPath = join(basePath, 'fuel.json')
+  logDebug(['fuel.json path:', fuelJSONPath], LogPrefix.Nile)
+
+  if (!existsSync(fuelJSONPath)) {
+    return null
+  }
+
+  try {
+    return JSON5.parse(readFileSync(fuelJSONPath, 'utf-8'))
+  } catch (error) {
+    logError(['Could not read', `${fuelJSONPath}:`, error], LogPrefix.Nile)
+  }
+
+  return null
+}
+
+/**
+ * Obtain a list of updateable games.
+ *
+ * @returns App names of updateable games.
+ */
+export async function listUpdateableGames(): Promise<string[]> {
+  if (!NileUser.isLoggedIn()) {
+    return []
+  }
+  logInfo('Looking for updates...', LogPrefix.Nile)
+
+  const abortID = 'nile-list-updates'
+  const { stdout: output } = await runRunnerCommand(
+    ['list-updates', '--json'],
+    createAbortController(abortID)
+  )
+  deleteAbortController(abortID)
+
+  if (!output) {
+    return []
+  }
+
+  const updates: string[] = JSON.parse(output)
+  if (updates.length) {
+    logInfo(['Found', `${updates.length}`, 'games to update'], LogPrefix.Nile)
+  }
+  return updates
+}
+
+/**
+ * Refresh games in the user's library
+ */
+async function refreshNile(): Promise<ExecResult> {
+  logInfo('Refreshing Amazon Games...', LogPrefix.Nile)
+
+  const abortID = 'nile-refresh'
+  const res = await runRunnerCommand(
+    ['library', 'sync'],
+    createAbortController(abortID)
+  )
+
+  deleteAbortController(abortID)
+
+  if (res.error) {
+    logError(['Failed to refresh library:', res.error], LogPrefix.Nile)
+  }
+  return {
+    stderr: '',
+    stdout: ''
+  }
+}
+
+export function getInstallMetadata(
+  appName: string
+): NileInstallMetadataInfo | undefined {
+  if (!existsSync(nileInstalled)) {
+    return
+  }
+
+  try {
+    const installed: NileInstallMetadataInfo[] = JSON.parse(
+      readFileSync(nileInstalled, 'utf-8')
+    )
+    return installed.find((game) => game.id === appName)
+  } catch (error) {
+    logError(
+      ['Corrupted installed.json file, cannot load installed games', error],
+      LogPrefix.Nile
+    )
+  }
+  return
+}
+
+/**
+ * Refresh `installedGames` from file.
+ */
+export function refreshInstalled() {
+  installedGames.clear()
+  if (existsSync(nileInstalled)) {
+    try {
+      const installed: NileInstallMetadataInfo[] = JSON.parse(
+        readFileSync(nileInstalled, 'utf-8')
+      )
+      installed.forEach((metadata) => {
+        installedGames.set(metadata.id, metadata)
+      })
+    } catch (error) {
+      logError(
+        ['Corrupted installed.json file, cannot load installed games', error],
+        LogPrefix.Nile
```

Comment:
I will take a look

Reply:
Showing an error on frontend is fine but at the same time we might have false positives here. Sometimes the file is fine but due to some re-render we might read it twice at the same time and this causes the error to appear. 
I know that because I added a frontend error message for that on Heroic at some point for Epic games and we were getting this error on every session at least once or twice. Especially after a game gets installed, because it updates the installed.json file, so if another refresh was called, this could cause the json to be incomplete while were being written and trigger the error.

I honestly think we should keep it as a log message and use it for debugging in case someone has some issues with installing games and so on here.

What do you think?

Reply:
yeah sounds good. thanks for the additional info

I didn't know we were having concurrency issues with read/writes to the json though? We could solve this using sqlite instead of json. I know it's a big lift, but we should really consider it if these are common errors. I think in some of the other store managers, it deletes the whole json if it detects that it's corrupted.

Could we add a sentry call in this catch black? 
like this
 ```
 import {captureException} from '@sentry/electron'
 
 ...
 
 catch (error) {
      logError(
        ['Corrupted installed.json file, cannot load installed games', error],
        LogPrefix.Nile
      )
     captureException(error)
 }
 ```